lib/translator/common.h

```diff
@@ -23,6 +27,14 @@ void DebugLog(const char* format, ...);
 
 void SetDebugCallback(void (*callback)(const char*));
 
+template <typename T>
+bool ReadValue(absl::Span<const uint8_t> data, T& out) {
+  static_assert(std::is_pod_v<T>, "only supports POD types");
+  if (sizeof(T) != data.size()) return false;
```

Comment:
I think we might find it useful in more scenarios to just copy as much data as possible into the structure "out" rather than enforcing the span be exactly of the same size. For example, imagine we have a large span but we know that the next 10 bytes is a particular structure that we want to cast.
So, if (sizeof(T) > data.size()) copySize = data.size();

Reply:
It's really easy to create a span for a prefix of another span (`foo.subspan(0, n)`), but I'm not sure requiring that boilerplate from callers adds any value, so I agree that being permissive here is sensible.

Reply:
That makes sense to me!